src/MonitoringDemo/Program.cs

```diff
@@ -0,0 +1,123 @@
+﻿namespace MonitoringDemo
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    class Program
+    {
+        static readonly CancellationTokenSource tokenSource = new CancellationTokenSource();
+
+        static async Task Main(string[] args)
+        {
+            var syncEvent = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            Console.CancelKeyPress += (sender, eventArgs) =>
+            {
+                eventArgs.Cancel = true;
+                tokenSource.Cancel();
+                syncEvent.TrySetResult(true);
+            };
+
+            try
+            {
+                using (var launcher = new DemoLauncher())
+                {
+                    Console.WriteLine("Starting the Particular Platform");
+
+                    launcher.Platform();
+
+                    using (ColoredConsole.Use(ConsoleColor.Yellow))
+                    {
+                        Console.WriteLine(
+                            "Once ServiceControl has finished starting a browser window will pop up showing the ServicePulse monitoring tab");
+                    }
+
+                    Console.WriteLine("Starting Demo Solution");
+
+                    if (!tokenSource.IsCancellationRequested)
+                    {
+                        Console.WriteLine("Starting Billing endpoint.");
+                        launcher.Billing();
+
+                        Console.WriteLine("Starting Sales endpoint.");
+                        launcher.Sales();
+
+                        Console.WriteLine("Starting Shipping endpoint.");
+                        launcher.Shipping();
+
+                        Console.WriteLine("Starting ClientUI endpoint.");
+                        launcher.ClientUI();
+
+                        using (ColoredConsole.Use(ConsoleColor.Yellow))
+                        {
+                            ScaleSalesEndpointIfRequired(launcher, syncEvent);
+
+                            await syncEvent.Task.ConfigureAwait(false);
+
+                            Console.WriteLine("Shutting down");
+                        }
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                using (ColoredConsole.Use(ConsoleColor.Red))
+                {
+                    Console.WriteLine("Error starting setting up demo.");
+                    Console.WriteLine($"{e.Message}{Environment.NewLine}{e.StackTrace}");
+                }
+            }
+
+            using (ColoredConsole.Use(ConsoleColor.Yellow))
+            {
+                Console.WriteLine("Done, press ENTER.");
+                Console.ReadLine();
+            }
+        }
+
+        private static void ScaleSalesEndpointIfRequired(DemoLauncher launcher, TaskCompletionSource<bool> syncEvent)

```

Comment:
How about we modify this part to be more user-friendly now that we have the proper language of expression and use something like `Press U to scale up the Sales service or D to scale it down`. And start/stop Sales instances based on this? I think it would be more fun.

Reply:
How about

`Press O to scale out the Sales service or I to scale in`